internal/degrees/degrees.go

```diff
@@ -0,0 +1,33 @@
+package degrees
+
+import (
+	"github.com/gzcharleszhang/course-planner/internal/courses"
+	"github.com/gzcharleszhang/course-planner/internal/students"
+)
+
+type DegreeName string
+type DegreeRequirements courses.CourseRequirementRules
+
+type Degree interface {
+	Plan(*courses.CoursePlan) bool
```

Comment:
What is `Plan` supposed to do?

Reply:
supposed to plan their future courses based on degrees/options/minors they declare, i guess it will be a future feature